### PR TITLE
#1919 SedStats Client changes

### PR DESCRIPTION
**Pull request by amedenblik**
Work on #1919

### CODE REVIEW

File: src/Controllers/SedimentController.ts [hunk 1]
```diff
@@ -90,6 +90,8 @@ module StreamStats.Controllers {
         private init(): void {
             this.isBusy = false;
 
+            this.selectedReferenceGage = new Models.ReferenceGage("", "");
```
Comment (amedenblik) [inline src/Controllers/SedimentController.ts:93]: First check to see if this.studyAreaService.selectedStudyArea.sedimentModelData.selectedGage is null or not

File: src/Controllers/SedimentController.ts [hunk 2]
```diff
@@ -382,8 +384,8 @@ module StreamStats.Controllers {
                 rg.Longitude_DD = gage.geometry.coordinates[1];
             }
             //add to list of reference gages
-            sid[0].options = this.referenceGageList;
-            sid[0].value = rg;
+            // sid[0].options = this.referenceGageList;
+            // sid[0].value = rg;
```
Comment (amedenblik) [inline src/Controllers/SedimentController.ts:388]: Not sure if these changes are necesary

File: src/Views/sedimentview.html [hunk 3]
```diff
@@ -102,7 +102,7 @@ <h5>Drainage area of clicked point: {{ vm.getDrainageArea() }}</h5>
                                     <td>{{gage.StartDate | date:'yyyy-MM-dd'}}</td>
                                     <td>{{gage.EndDate | date:'yyyy-MM-dd'}}</td>
                                     <td>
-                                        <a ng-class="{linkDisabled: gage.StationID == vm.selectedReferenceGage.StationID || !vm.checkCorrelationMatrix(gage) || !vm.checkPeriodOfRecord(gage) || !gage.HasFlowDurationStats || gage.stationType.code != 'GS'}" ng-click="vm.selectGage(gage)">
+                                        <a ng-class="{linkDisabled: gage.StationID == vm.selectedReferenceGage.StationID || !vm.checkPeriodOfRecord(gage) || gage.stationType.code != 'GS'}" ng-click="vm.selectGage(gage)">
```
Comment (amedenblik) [inline src/Views/sedimentview.html:105]: vm.checkCorrelationMatrix(gage) and gage.HasFlowDurationStats aren't relevant for this model

File: src/Services/StudyAreaService.ts [hunk 4]
```diff
@@ -73,6 +73,7 @@ module StreamStats.Services {
         extensionDateRange: IDateRange;
         selectedGage: any;        
         flowAnywhereData: any;
+        sedimentModelData: any;
```
Comment (amedenblik) [inline src/Services/StudyAreaService.ts:76]: Create an object (optionally with interface) to keep track of sediment model data from sediment modal (date range and selected gage)

File: src/Services/StudyAreaService.ts [hunk 5]
```diff
@@ -1009,6 +1010,9 @@ module StreamStats.Services {
 
         public selectGage(gage) {
             // selects gage and adds it to gage options for qppq
+            if () {
```
Comment (amedenblik) [inline src/Services/StudyAreaService.ts:1013]: If the applications includes Sediment, do the stuff for sediment. Else if the applications includes FDCTM, use the existing code in this function.

**Comment by amedenblik** [comment]
@njestes feel free to either merge this PR or close it, whatever is most helpful for you!

**Review by njestes — APPROVED** [review]

**Pull request merged**